spyder/utils/ipython/spyder_kernel.py

```diff
@@ -212,11 +214,12 @@ def publish_pdb_state(self):
         Publish Variable Explorer state and Pdb step through
         publish_data.
         """
-        if self._pdb_obj:
+        if self._pdb_obj and not self._publish_data:
             state = dict(namespace_view = self.get_namespace_view(),
                          var_properties = self.get_var_properties(),
                          step = self._pdb_step)
             publish_data({'__spy_pdb_state__': state})
+        self._publish_data = False
```

Comment:
What happens if you open several variables while debugging? I think then this approach wouldn't work all the time.

Reply:
Do you mean like a synchronization issue?. A preview:

![debug](https://user-images.githubusercontent.com/16781833/31962127-50539c4c-b8c2-11e7-8cbf-e4d720759cec.gif)

Reply:
Nice, it's working as expected!